templates/bond_nmconnection.j2

```diff
@@ -101,7 +101,18 @@ routing-rule{{ loop.index }}=priority {{ 32765 - loop.index0 }} {{ rule }}
 {%   endif %}
 {% endfor %}
 {% if item.dnsnameservers is defined %}
-dns={{ item.dnsnameservers | join(',') }}
+{% if item.dnsnameservers is iterable %}
```

Comment:
Unfortunately this test returns true for strings:

```
ansible localhost -m debug -a msg='{{ "abc" is iterable }}'

[WARNING]: No inventory was parsed, only implicit localhost is available
localhost | SUCCESS => {
    "msg": true
}
```

How about `is not string`?

Reply:
Also [this post](https://superuser.com/a/606547) suggests that commas are correct. What is your source for using semicolons?

Reply:
That the interfaces did not update on RHEL9 when using commas.

Reply:
Your reference is for regular `ifcfg` files, here is the reference for `nmconnection` https://developer-old.gnome.org/NetworkManager/stable/nm-settings-keyfile.html

Reply:
I've upated the test to `is not string` and tested it